drivers/bridge/bridge.go

```diff
@@ -245,35 +272,70 @@ func (d *driver) CreateNetwork(id types.UUID, option map[string]interface{}) err
 	d.Lock()
 
 	// Sanity checks
-	if d.network != nil {
+	if _, ok := d.networks[id]; ok {
 		d.Unlock()
-		return &ErrNetworkExists{}
+		return types.ForbiddenErrorf("Network %s exists", id)
+	}
+
+	// Parse and validate the config. It should not conflict with existing networks' config
+	config, err := parseNetworkOptions(option)
+	if err != nil {
+		d.Unlock()
+		return err
+	}
+	for _, nw := range d.networks {
+		if nw.config.Conflict(config) {
+			d.Unlock()
+			return types.ForbiddenErrorf("Conflicts with network %s (%s)", nw.id, nw.config.BridgeName)
+		}
 	}
 
 	// Create and set network handler in driver
-	d.network = &bridgeNetwork{id: id, endpoints: make(map[types.UUID]*bridgeEndpoint)}
-	network := d.network
+	network := &bridgeNetwork{id: id, endpoints: make(map[types.UUID]*bridgeEndpoint), config: config}
+	d.networks[id] = network
 	d.Unlock()
 
 	// On failure make sure to reset driver network handler to nil
 	defer func() {
 		if err != nil {
 			d.Lock()
-			d.network = nil
+			delete(d.networks, id)
 			d.Unlock()
 		}
 	}()
 
-	config, err := parseNetworkOptions(option)
-	if err != nil {
-		return err
-	}
-	network.config = config
-
 	// Create or retrieve the bridge L3 interface
 	bridgeIface := newInterface(config)
 	network.bridge = bridgeIface
 
+	// Verify network does not conflict with previously configured networks
+	// on parameters that were chosen by the driver.
+	d.Lock()
+	for _, nw := range d.networks {
+		if nw.id != id {
+			// Verify the name (which may have been set by newInterface()) does not conflict with
+			// existing bridge interfaces. Ironically the system chosen name gets stored in the config...
+			// Basically we are checking if the two original configs were both empty.
+			if nw.config.BridgeName == config.BridgeName {
+				d.Unlock()
+				return types.ForbiddenErrorf("Conflicts with network %s (%s)", nw.id, nw.config.BridgeName)
+			}
+			// If this network config specifies the AddressIPv4, we need
+			// to make sure it does not conflict with any previously allocated
+			// bridges. This could not be completely caught by the config conflict
+			// check, because networks which config does not specify the AddressIPv4
+			// get their address and subnet selected by the driver (see electBridgeIPv4())
+			if nw.config.AddressIPv4 != nil {
```

Comment:
So there is still is a bug here, the if check should be on the current network being configured, not the already programmed ones. Thanks.